scripts/data/utxo_processor.py

```diff
@@ -0,0 +1,111 @@
+    os.makedirs(file_dir, exist_ok=True)
+
+    url = f"{GCS_BASE_URL}{file_name}"
+    response = requests.get(url)
+    if response.status_code != 200:
+        raise Exception(f"Failed to download {file_name}")
+
+    file_path = os.path.join(BASE_DIR, file_name)
+    with open(file_path, "wb") as f:
+        f.write(response.content)
+
+    # Split file
+    split_cmd = f"split -l {CHUNK_SIZE} {file_path} {file_dir}/chunk_"
+    subprocess.run(split_cmd, shell=True, check=True)
+
+    # Remove original file
+    os.remove(file_path)
+
+
+def create_index():
+    """Create an index mapping block numbers to chunk files."""
+    index: Dict[int, str] = {}
+
+    for dir_name in os.listdir(BASE_DIR):
+        dir_path = os.path.join(BASE_DIR, dir_name)
+        if not os.path.isdir(dir_path):
+            continue
+
+        for chunk_file in os.listdir(dir_path):
+            chunk_path = os.path.join(dir_path, chunk_file)
+            with open(chunk_path, "r") as f:
+                for line_num, line in enumerate(f, 1):
+                    try:
+                        data = json.loads(line.strip())
+                        block_number = data["block_number"]
+                        if block_number in index:
+                            print(f"Warning: Duplicate block number {block_number}")
+                        index[block_number] = chunk_path
+                    except json.JSONDecodeError as e:
+                        print(
+                            f"Error decoding JSON in file {chunk_path}, line {line_num}: {e}"
+                        )
+                        print(f"Problematic line: {line.strip()}")
+                    except KeyError as e:
+                        print(f"KeyError in file {chunk_path}, line {line_num}: {e}")
+                        print(f"Data: {data}")
+
+    with open(INDEX_FILE, "w") as f:
+        json.dump(index, f)
+
+    print(f"Index created with {len(index)} entries")
+
+
+def get_utxo_set(block_number: int) -> Dict[str, Any]:
+    """Get the UTXO set for a given block number."""
+    # Load index
+    if not os.path.exists(INDEX_FILE):
+        create_index()
+
+    with open(INDEX_FILE, "r") as f:
+        index = json.load(f)
+
+    # Find chunk file
+    chunk_file = index.get(str(block_number))
+    if not chunk_file:
+        raise Exception(f"Block number {block_number} not found in index")
+
+    # If chunk file doesn't exist, download and split
+    if not os.path.exists(chunk_file):
+        file_name = os.path.basename(os.path.dirname(chunk_file)) + ".json"
+        download_and_split(file_name)
+
+    # Find and return data for the block
+    with open(chunk_file, "r") as f:
+        for line in f:
+            data = json.loads(line.strip())
+            if data["block_number"] == block_number:
+                return data
+
+    raise Exception(f"Block {block_number} not found in chunk file {chunk_file}")
+
+
+# Example usage
+if __name__ == "__main__":
```

Comment:
Please make main accept a range of files to process, for example '--from 000...01 --to 000...50'.


It should also report progress to the console.

Reply:
<img width="1095" alt="Screenshot 2024-09-23 at 7 12 56 AM" src="https://github.com/user-attachments/assets/a9a49df6-e35a-4590-b4d4-67fb74a160d9">
Updated @maciejka

Reply:
<img width="1082" alt="Screenshot 2024-09-23 at 7 26 11 AM" src="https://github.com/user-attachments/assets/ade41f82-c5d8-4ad5-8223-79766ff92fcb">

All tests passed too.